src/storage/src/source/mysql/snapshot.rs

```diff
@@ -362,6 +362,51 @@ pub(crate) fn render<G: Scope<Timestamp = GtidPartition>>(
                     }
                 }
 
+                // Verify the schemas of the tables we are snapshotting
+                let cur_schemas: BTreeMap<_, _> = schema_info(
+                    &mut conn,
+                    &SchemaRequest::Tables(
+                        reader_snapshot_table_info
+                            .keys()
+                            .map(|f| (f.0[0].as_str(), f.0[1].as_str()))
+                            .collect(),
+                    ),
+                )
+                .await?
+                .drain(..)
+                .map(|schema| {
+                    (
+                        table_name(&schema.schema_name, &schema.name).unwrap(),
+                        schema,
+                    )
+                })
+                .collect();
+                let mut removed_tables = vec![];
+                for (table, (output_index, table_desc)) in reader_snapshot_table_info.iter() {
+                    if let Err(err) = verify_schema(table, &table_desc, &cur_schemas) {
+                        // Put this table into an error state and stop ingesting it
+                        return_definite_error(
+                            err,
+                            &[*output_index],
+                            &mut raw_handle,
+                            data_cap_set,
+                            &mut definite_error_handle,
+                            definite_error_cap_set,
+                        )
+                        .await;
+                        trace!(%id, "timely-{worker_id} stopping snapshot of table {table} \
+                                     due to schema mismatch");
+                        removed_tables.push(table.clone());
+                    }
+                }
+                for table in &removed_tables {
+                    reader_snapshot_table_info.remove(table);
+                }
+                // This worker has nothing else to do
+                if reader_snapshot_table_info.is_empty() {
+                    return Ok(());
+                }
+
                 // We have established a snapshot frontier so we can broadcast the rewind requests
```

Comment:
I moved the schema checks above the rewind requests since it seems that we don't need to issue a rewind for a table we don't care about.

But it's worth noting that in the Postgres source we do the schema checks after we've already published a rewind request for this table, which will eventually get dropped in the replication operator but might unnecessarily hold a capability for a table we won't actually be ingesting.